specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json

```diff
@@ -0,0 +1,212 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Trunk health Service",
+    "description": "Trunk health Service",
+    "contact": {
+      "email": "acsdevexdisc@microsoft.com"
+    },
+    "version": "2022-07-01-preview1"
+  },
+  "paths": {
```

Comment:
Overall, this new API seems a bit too narrow to me, especially in the following context:

`/sip` already has a GET method. The response to that includes a map of `trunks`. The map there contains trunk FQDN (as the key of the map) and the `sipSignalingPort`.
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/communication/data-plane/SipRouting/preview/2021-05-01-preview/communicationservicessiprouting.json

Now a part of the information about the trunk is split to this separate API under `sip/trunkHealth`.

Two desires:
* Keep information about trunks in one place instead of two.
* Avoid a narrow api path that only stores health information for trunks.

There are a few options to meet these desires:
* Do not introduce the `trunkHealths` collection, instead, expand the `trunks` map under `/sip` and depend on JSON merge-path etc.
  * Should only do this if the set of `trunkHealths` changes a lot, and/or is limited in size.
* Introduce `/sip/trunks` and move the FQDN and `sipSignalingPort` in there. Drop the map under `/sip`
* Introduce `/sip/trunks` to contain static information (like `sipSignalingPort` and `fqdn`) and `sip/trunks/<fqdn>/health` to get the dynamic health for a trunk. The problem here is that it's harder to get the health for *all* trunks.

There is the extra complication that health information is changed by the server, while `sipSignalingPort` is changed by the client (?). This could be confusing. The last option (and the current proposal) mitigates this by keeping the two separate.

--

In any case, splitting the trunk information across a map and a collection seems weird because maps and collections fit different requirements:
e.g, collections must support pagination, with the assumption that collection can grow to be large. In this case, having related information in a map becomes a problem.

Reply:
FYI related suggestion from a previous review of this api: #19376

Reply:
I recall there was also a suggestion to use `expand`: `/sip/trunks?expand=healthStatus` or `/sip?expand=healthStatus` - did we consider this?

Reply:
As agreed on internal review, we will expand `trunks` map under `/sip?expand=trunkHealth`